src/devices/Button/GpioButton.cs

```diff
@@ -66,8 +72,19 @@ public class GpioButton : ButtonBase
                 ? _gpioPinMode = PinMode.Input
                 : _gpioPinMode = _eventPinMode;
 
-            _gpioController.OpenPin(_buttonPin, _gpioPinMode);
-            _gpioController.RegisterCallbackForPinValueChangedEvent(_buttonPin, PinEventTypes.Falling | PinEventTypes.Rising, PinStateChanged);
+            try
+            {
+                _gpioController.OpenPin(_buttonPin, _gpioPinMode);
+                _gpioController.RegisterCallbackForPinValueChangedEvent(
+                    _buttonPin,
+                    PinEventTypes.Falling | PinEventTypes.Rising,
+                    PinStateChanged);
+            }
+            catch (Exception)
+            {
+                _gpioController.Dispose();
```

Comment:
Almost perfect. Add `if (_shouldDispose)` here. (For some reason, I cannot post suggestions.)